src/inmanta/parser/plyInmantaParser.py

```diff
@@ -205,7 +220,16 @@ def p_stmt(p: YaccProduction) -> None:
     """statement : assign
     | for
     | if
-    | expression empty"""
+    | boolean_expression
+    | constant
+    | function_call
+    | var_ref empty
+    | constructor
+    | map_def
+    | map_lookup empty
+    | index_lookup
+    | conditional_expression
+    | arithmetic_expression"""  # Can't do anything starting with `[` as it breaks dict lookup
```

Comment:
This is the nasty one: can't have expression starting with `[` 

because

```
a = b
['a']
```
can either be 

```
Assign('a', Reference('b'))
CreateLis([Literal('a')])
```
or
```
Assign('a', MapLookup(Reference('b'),'a'))
```

Reply:
Wasn't this also the case before? i.e. on master:
```
d = {"a": 42}

x = d
["a"]


std::print(x)
```

produces 42. It's just as ambiguous.